apps/coscrad-frontend/src/components/higher-order-components/categorizable-page-layout.tsx

```diff
@@ -0,0 +1,174 @@
+import { AggregateCompositeIdentifier } from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    Close as CloseIcon,
+    ExpandLess as ExpandLessIcon,
+    ExpandMore as ExpandMoreIcon,
+    Hub as HubIcon,
+    TextSnippet as TextSnippetIcon,
+} from '@mui/icons-material';
+import {
+    Box,
+    Divider,
+    Drawer,
+    IconButton,
+    Stack,
+    Tooltip,
+    Typography,
+    styled,
+} from '@mui/material';
+import { ReactNode, useState } from 'react';
+import { CommandPanel } from '../commands';
+
+interface CategorizablePageLayoutProps {
+    compositeIdentifier: AggregateCompositeIdentifier;
+    selfNotesList: JSX.Element;
+    connectedResourcesList: JSX.Element;
+    commandPanel?: JSX.Element;
+    children: ReactNode;
+}
+
+const DrawerHeader = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    alignItems: 'flex-start',
+    margin: theme.spacing(2, 4, 0, 4),
+    justifyContent: 'space-between',
+}));
+
+const DrawerContentStack = styled(Stack)(({ theme }) => ({
+    margin: theme.spacing(0, 4, 0, 4),
+    overflow: 'scroll',
+}));
+
+type Anchor = 'top' | 'left' | 'bottom' | 'right';
+
+export const CategorizablePageLayout = ({
+    compositeIdentifier: { type: resourceType, id },
+    selfNotesList,
+    connectedResourcesList,
+    commandPanel,
+    children,
+}: CategorizablePageLayoutProps) => {
+    const [drawerState, setDrawerState] = useState({
+        bottom: false,
+        right: false,
+    });
+    const [isBottomDrawerExpanded, setIsBottomDrawerExpanded] = useState(false);
+
+    const toggleDrawer = (anchor: Anchor, isOpen: boolean) => {
+        setDrawerState({ ...drawerState, [anchor]: isOpen });
```

Comment:
Is this really what you want to do? anchor can take on one of 4 values but you initialize your state with only 2 possible keys. If any of the 4 strings in the `Anchor` union is allowed as a key, you need to initialize all 4. However, this suffers from the design defect that either all 4 can be open for once or you need a complex layer for managing the rule that not all can be open at once.